loaders/src/glTF/2.0/glTFLoader.ts

```diff
@@ -1675,8 +1676,24 @@ export class GLTFLoader implements IGLTFLoader {
             const bufferView = ArrayItem.Get(`${context}/bufferView`, this._gltf.bufferViews, accessor.bufferView);
             accessor._babylonVertexBuffer = this._loadVertexBufferViewAsync(bufferView, kind).then((babylonBuffer) => {
                 const size = GLTFLoader._GetNumComponents(context, accessor.type);
-                return new VertexBuffer(this._babylonScene.getEngine(), babylonBuffer, kind, false, false, bufferView.byteStride,
+                var vertexBuffer = new VertexBuffer(this._babylonScene.getEngine(), babylonBuffer, kind, false, false, bufferView.byteStride,
                     false, accessor.byteOffset, size, accessor.componentType, accessor.normalized, true);
+
+                // gltfload owns this buffer and is responsible to dispose it at some point.
+                // for now, it's doing that procedure when the engine is disposed
+                if (babylonBuffer instanceof Buffer) {
+                    GLTFLoader._buffersToDispose.push(vertexBuffer);
+                    this._babylonScene.getEngine().onDispose = () => {
```

Comment:
Engine dispose seems too coarse grained. Can we do this on scene scene dispose?

Reply:
correct me if I'm wrong @bghgary but assets loading with gltf belong to an asset container. and that asset container can be used to clone assets to different scenes. 1st scene dispose would result in invalid buffer for the other ones.

Reply:
asset container lives within a scene so we are good at the scene level (and it is better for memory pressure as well)

Reply:
Furthermore this will help not adding code to thinengine